MobileCenter/MobileCenter/Internals/Sender/MSIngestionSender.m

```diff
@@ -10,28 +10,31 @@ @implementation MSIngestionSender
 static NSString *const kMSAPIVersionKey = @"api_version";
 static NSString *const kMSApiPath = @"/logs";
 
-- (id)initWithBaseUrl:(NSString *)baseUrl
-              appSecret:(NSString *)appSecret
-              installId:(NSString *)installId {
+- (id)initWithBaseUrl:(NSString *)baseUrl appSecret:(NSString *)appSecret installId:(NSString *)installId {
   self = [super initWithBaseUrl:baseUrl
-                        apiPath:kMSApiPath
-                        headers:@{
-                                  kMSHeaderContentTypeKey : kMSContentType,
-                                  kMSHeaderAppSecretKey : appSecret,
-                                  kMSHeaderInstallIDKey : installId
-                                  }
-                   queryStrings:@{
-                                  kMSAPIVersionKey : kMSAPIVersion
-                                  }
-                   reachability:[MS_Reachability reachabilityForInternetConnection]
-                 retryIntervals:@[ @(10), @(5 * 60), @(20 * 60) ]];
+      apiPath:kMSApiPath
+      headers:@{
+        kMSHeaderContentTypeKey : kMSContentType,
+        kMSHeaderAppSecretKey : appSecret,
+        kMSHeaderInstallIDKey : installId
+      }
+      queryStrings:@{
+        kMSAPIVersionKey : kMSAPIVersion
+      }
+      reachability:[MS_Reachability reachabilityForInternetConnection]
+      retryIntervals:@[ @(10), @(5 * 60), @(20 * 60) ]];
   return self;
 }
 
 - (void)sendAsync:(NSObject *)data completionHandler:(MSSendAsyncCompletionHandler)handler {
   MSLogContainer *container = (MSLogContainer *)data;
   NSString *batchId = container.batchId;
 
+  /*
+   * FIXME: All logs are already validated at the time the logs are enqueued to Channel. It is not necessary but it can
+   * still protect against invalid logs being sent to server that are messed up somehow in Storage. If we see
+   * performance issues due to this validation, we will remove `[container isValid]` call below.
+   */
```

Comment:
Have you tried profiling one or our sample apps? I'd rather invest 1-2hrs into checking the performance impact of validating everything rather than having to deal with this again.